data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/coordination/partition/ResyncPartition.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.source.rds.coordination.partition;
+
+import org.opensearch.dataprepper.model.source.coordinator.SourcePartitionStoreItem;
+import org.opensearch.dataprepper.model.source.coordinator.enhanced.EnhancedSourcePartition;
+import org.opensearch.dataprepper.plugins.source.rds.coordination.state.ResyncProgressState;
+
+import java.util.Optional;
+
+public class ResyncPartition extends EnhancedSourcePartition<ResyncProgressState> {
+
+    public static final String PARTITION_TYPE = "RESYNC";
+
+    private final String database;
+    private final String table;
+    private final long timestamp;
```

Comment:
Multiple update on different rows can happen at same timestamp. How is this been handled ? Should we also have some representation of row (primary key) here ?

Reply:
The timestamp is the timestamp on the binlog events from parent table. So the handling is similar to stream events.